### PR TITLE
[SWE-1463][BRANDBOOK MODULE] Exterior screen component #done

### PR DESCRIPTION
**Pull request by ArtemSuhoy**

### CODE REVIEW

File: src/app/components/bundles/GreetingScreen/components/GreetingScreenInfo/GreetingScreenInfo.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { Col } from 'components'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+const GreetingScreenInfo = (props) => {
+  const { title } = props
+  return (
+    <Col cw={12}>
+      <h5>{title}</h5>
+      <div>Divider</div>
```
Comment (olenashevchuk) [inline src/app/components/bundles/GreetingScreen/components/GreetingScreenInfo/GreetingScreenInfo.js:10]: you didn`t see workshop

File: src/app/components/bundles/GreetingScreen/GreetingScreen.js [hunk 2]
```diff
@@ -1,7 +1,25 @@
...
+
+import PropTypes from 'prop-types'
 import React from 'react'
 
-const GreetingScreen = () => {
-  return <></>
+const GreetingScreen = (props) => {
+  const { title, content } = props
+  return (
+    <GreetingScreenContainer>
```
Comment (olenashevchuk) [inline src/app/components/bundles/GreetingScreen/GreetingScreen.js:13]: add Header and Footer

**Review by olenashevchuk — CHANGES_REQUESTED** [review]

File: src/app/components/bundles/GreetingScreen/GreetingScreen.js [hunk 3]
```diff
@@ -1,7 +1,25 @@
...
+  GreetingScreenInfo
+} from './components'
+
+import PropTypes from 'prop-types'
 import React from 'react'
 
-const GreetingScreen = () => {
-  return <></>
+const GreetingScreen = (props) => {
+  const { title, content } = props
```
Comment (Olexandr25) [inline src/app/components/bundles/GreetingScreen/GreetingScreen.js:11]: add comment, what this component
- do
- what type of props, and for what
-

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          <PositionedContainer bottom="0">
+            <Divider color="accent" bold size="sm" />
+          </PositionedContainer>
+        </Box>
+      </Box>
+    </Col>
+  )
+}
+
+ExteriorScreenInfo.propTypes = {
```
Comment (olenashevchuk) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js:38]: add prop types

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          {computedTitle}
+          <Title color="accent" variant="heading2">
+            {lastCharacterTitle}
+          </Title>
+        </Title>
+      </Box>
+
+      <Box pl={'64px'}>
+        <Box pl={'32px'}>
+          <Text variant="subtitle2">{subtitle}</Text>
```
Comment (olenashevchuk) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js:25]: strange

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+const ExteriorScreenInfo = (props) => {
+  const { subtitle, title } = props
+  /* [COMPUTER_PROPERTIES] */
+  const computedTitle = `${title.slice(0, title.indexOf('.'))}`
+  const lastCharacterTitle = title[title.length - 1]
+
+  return (
+    <Col cw={11} v="bottom">
+      <Box pb={'32px'}>
```
Comment (olenashevchuk) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js:14]: use mb={4} instead pb={'32px'}>

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+import React from 'react'
+
+const ExteriorScreenInfo = (props) => {
+  const { subtitle, title } = props
+  /* [COMPUTER_PROPERTIES] */
+  const computedTitle = `${title.slice(0, title.indexOf('.'))}`
+  const lastCharacterTitle = title[title.length - 1]
+
+  return (
+    <Col cw={11} v="bottom">
```
Comment (olenashevchuk) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js:13]: why 11? there is 10

File: src/app/components/elements/Media/Img/Img.styled.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import styled from 'styled-components'
+
+const ImgStyled = styled.img`
+  position: ${({ position }) => position};
+  left: ${({ left }) => left};
+  right: ${({ right }) => right};
+  top: ${({ top }) => top};
+  bottom: ${({ bottom }) => bottom};
+  z-index: ${({ zIndex }) => zIndex};
+`
```
Comment (olenashevchuk) [inline src/app/components/elements/Media/Img/Img.styled.js:10]: wrong

File: src/app/screens/Default/ClosingScreen/ClosingScreen.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import { ExteriorScreen } from 'components'
+
+const ClosingScreen = () => {
+  return (
+    <ExteriorScreen
+      title={'Thank you for watching!'}
```
Comment (olenashevchuk) [inline src/app/screens/Default/ClosingScreen/ClosingScreen.js:6]: delete {}

File: src/app/screens/Default/OpeningScreen/OpeningScreen.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import { ExteriorScreen } from 'components'
+
+const OpeningScreen = () => {
+  return (
+    <ExteriorScreen
+      title={'Brand guidelines.'}
+      subtitle="Summer 2022"
+      decoration={'opening'}
+    />
```
Comment (olenashevchuk) [inline src/app/screens/Default/OpeningScreen/OpeningScreen.js:9]: delete {}

**Review by olenashevchuk — CHANGES_REQUESTED** [review]

**Review by olenashevchuk — APPROVED** [review]

File: src/app/components/bundles/ExteriorScreen/ExteriorScreen.js [hunk 11]
```diff
@@ -1,7 +1,53 @@
...
+
+import ExteriorScreenStyled from './ExteriorScreen.styled'
+import PropTypes from 'prop-types'
 import React from 'react'
 
-const ExteriorScreen = () => {
-  return <></>
+const ExteriorScreen = (props) => {
+  /**
+   *  Used for first and last screen -- greeting and thanks
```
Comment (Olexandr25) [inline src/app/components/bundles/ExteriorScreen/ExteriorScreen.js:10]: move comment above 
const ExteriorScreen = (prop) => {

File: src/app/components/bundles/ExteriorScreen/ExteriorScreen.js [hunk 12]
```diff
@@ -1,7 +1,53 @@
...
+   *  @param subtitle - subtitle for information
+   *  @param title - main title page
+   *  @param decoration - closing or opening, choose background to display
+   */
+
+  const { subtitle, title, decoration } = props
+
+  /* [COMPUTED_PROPERTIES] */
+  const decorationBgProps = {
```
Comment (Olexandr25) [inline src/app/components/bundles/ExteriorScreen/ExteriorScreen.js:19]: move to __constants__

File: src/app/components/bundles/ExteriorScreen/ExteriorScreen.js [hunk 13]
```diff
@@ -1,7 +1,53 @@
...
+      left: '360px',
+      top: '-88px'
+    },
+    closing: {
+      left: '216px',
+      top: '48px'
+    }
+  }
+
+  const imageSrc = decoration === 'opening' ? 'ill-bg.svg' : 'ill-thx.svg'
```
Comment (Olexandr25) [inline src/app/components/bundles/ExteriorScreen/ExteriorScreen.js:30]: leave comment, Why and For what result you do it

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenContainer/ExteriorScreenContainer.js [hunk 14]
```diff
@@ -0,0 +1,20 @@
+import ExteriorScreenContainerStyled from './ExteriorScreenContainer.styled'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { Row } from 'components'
+
+const ExteriorScreenContainer = (props) => {
```
Comment (Olexandr25) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenContainer/ExteriorScreenContainer.js:6]: write comment - js doc

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js [hunk 15]
```diff
@@ -0,0 +1,44 @@
+import { Box, Col, Divider, PositionedContainer, Text, Title } from 'components'
+
+import PropTypes from 'prop-types'
+import React from 'react'
+
+const ExteriorScreenInfo = (props) => {
```
Comment (Olexandr25) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js:6]: comment - js doc

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js [hunk 16]
```diff
@@ -0,0 +1,44 @@
+import { Box, Col, Divider, PositionedContainer, Text, Title } from 'components'
+
+import PropTypes from 'prop-types'
+import React from 'react'
```
Comment (Olexandr25) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js:4]: do you need React here?

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/index.js [hunk 17]
```diff
@@ -0,0 +1 @@
+export { default } from './ExteriorScreenInfo'
```
Comment (Olexandr25) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/index.js:1]: change export

File: src/app/components/elements/Media/Img/Img.js [hunk 18]
```diff
@@ -1,7 +1,12 @@
+import PropTypes from 'prop-types'
 import React from 'react'
 
-const Img = () => {
-  return <></>
+const Img = (props) => {
```
Comment (Olexandr25) [inline src/app/components/elements/Media/Img/Img.js:4]: comment - js doc

File: src/app/screens/Default/ClosingScreen/ClosingScreen.js [hunk 19]
```diff
@@ -0,0 +1,13 @@
+import { ExteriorScreen } from 'components'
+
+const ClosingScreen = () => {
```
Comment (Olexandr25) [inline src/app/screens/Default/ClosingScreen/ClosingScreen.js:3]: leave comment - for what this component

File: src/app/screens/Default/ClosingScreen/index.js [hunk 20]
```diff
@@ -0,0 +1 @@
+export { default } from './ClosingScreen'
```
Comment (Olexandr25) [inline src/app/screens/Default/ClosingScreen/index.js:1]: check this export, it is a little strange

File: src/app/screens/Default/OpeningScreen/OpeningScreen.js [hunk 21]
```diff
@@ -0,0 +1,13 @@
+import { ExteriorScreen } from 'components'
+
+const OpeningScreen = () => {
```
Comment (Olexandr25) [inline src/app/screens/Default/OpeningScreen/OpeningScreen.js:3]: leave comment - what this component return

File: src/app/components/bundles/ExteriorScreen/ExteriorScreen.js [hunk 22]
```diff
@@ -1,7 +1,54 @@
...
+      top: '48px'
+    }
+  }
+
+  // Checks the screen type and inserts the necessary decorations
+  const imageSrc = decoration === 'opening' ? 'ill-bg.svg' : 'ill-thx.svg'
+
+  return (
+    <ExteriorScreenStyled>
+      <PositionedContainer {...decorationBgProps[decoration]} zIndex="-10">
```
Comment (alexpassss) [inline src/app/components/bundles/ExteriorScreen/ExteriorScreen.js:35]: move zIndex into the decorationBgProps

put decorationBgProps outside of the ExteriorScreen function

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenContainer/ExteriorScreenContainer.js [hunk 23]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ *  Used for first and last screen -- greeting and thanks
+ *  Container for children components
+ *  @param children - Holds in ScreenInfo
+ */
+const ExteriorScreenContainer = (props) => {
+  const { children } = props
+  return (
+    <ExteriorScreenContainerStyled>
+      <Row height="100%" alignContent="center">
```
Comment (alexpassss) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenContainer/ExteriorScreenContainer.js:15]: bad practice to use Row without Col right after

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+
+const ExteriorScreenInfo = (props) => {
+  const { subtitle, title } = props
+
+  /* [COMPUTER_PROPERTIES] */
+  const computedTitle = `${title.slice(0, title.indexOf('.'))}`
+  const lastCharacterTitle = title[title.length - 1]
+
+  return (
+    <Col cw={10} v="bottom">
```
Comment (alexpassss) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js:18]: Bad practice to use Col without Row

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js [hunk 25]
```diff
@@ -0,0 +1,48 @@
...
+      <Box pb={4}>
```
Comment (alexpassss) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js:19]: better to use margin for such cases

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js [hunk 26]
```diff
@@ -0,0 +1,48 @@
...
+        <Title variant="heading2">
+          {computedTitle}
+          <Title color="accent" variant="heading2">
```
Comment (alexpassss) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js:22]: bad practice to use H tag inside H tag. Use styled Text instead

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js [hunk 27]
```diff
@@ -0,0 +1,48 @@
...
+            {lastCharacterTitle}
+          </Title>
+        </Title>
+      </Box>
+
+      <Box pl={5}>
```
Comment (alexpassss) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js:28]: better to use margin

File: src/app/components/elements/Media/Img/Img.js [hunk 28]
```diff
@@ -1,7 +1,17 @@
...
+ * @param src - link to picture
+ * @param alt - when image not loaded
+ * @returns image
+ */
+const Img = (props) => {
+  const { src, alt } = props
+  return <img src={src} alt={alt} />
+}
+Img.propTypes = {
+  src: PropTypes.string,
```
Comment (alexpassss) [inline src/app/components/elements/Media/Img/Img.js:14]: it should be required

File: src/app/components/elements/Media/Img/Img.js [hunk 29]
```diff
@@ -1,7 +1,17 @@
...
+  alt: PropTypes.string
```
Comment (alexpassss) [inline src/app/components/elements/Media/Img/Img.js:15]: it should be required

File: src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js [hunk 30]
```diff
@@ -0,0 +1,48 @@
...
+            <Divider color="accent" bold size="sm" />
+          </PositionedContainer>
+        </Box>
+      </Box>
+    </Col>
+  )
+}
+
+ExteriorScreenInfo.propTypes = {
+  title: PropTypes.string,
```
Comment (alexpassss) [inline src/app/components/bundles/ExteriorScreen/components/ExteriorScreenInfo/ExteriorScreenInfo.js:44]: it should be required

File: src/app/components/bundles/ExteriorScreen/ExteriorScreen.js [hunk 31]
```diff
@@ -1,7 +1,49 @@
...
+const ExteriorScreen = (props) => {
+  const { subtitle, title, decoration } = props
+
+  /* [COMPUTED_PROPERTIES] */
+
+  // Checks the screen type and inserts the necessary decorations
+  const imageSrc = decoration === 'opening' ? 'ill-bg.svg' : 'ill-thx.svg'
+
+  return (
+    <ExteriorScreenStyled>
```
Comment (alexpassss) [inline src/app/components/bundles/ExteriorScreen/ExteriorScreen.js:29]: Seems that this component will be reusable across all screens

**Pull request merged**